csrc/optimization/mark_aliases_prepare.cpp

```diff
@@ -28,37 +28,37 @@ void MarkAliasesPreparePass::runPass(Fusion* fusion) {
   // an input). Code will later add `segment_set` before them so aliases are
   // separated from non-aliases and more likely to be accepted by the no-op
   // scheduler.
-  std::vector<TensorView*> aliased_outs;
-  aliased_outs.reserve(fusion->outputs().size());
+  std::unordered_set<TensorView*> aliased_outs;
 
-  for (TensorView* out :
-       ir_utils::filterByType<TensorView>(fusion->outputs())) {
-    TensorView* aliased_io = analysis.getNearestAliasedIo(out);
+  for (TensorView* tv : ir_utils::allTvs(fusion)) {
```

Comment:
QQ: what's the motivation of marking alias on intermediate tensor here?

Is that to accommodate segmentation where intermediate could become another global buffer?

I was under the impression that the alias analysis thing would run again after segmentation. Not arguing that should have been the case though.

Reply:
Got'ya.

Thanks for the explanation.